### PR TITLE
Remove polaris-react/src/components/**/README.md files

### PR DESCRIPTION
**Pull request by alex-page**
**To ship first**
- [x] https://github.com/Shopify/polaris/pull/6771
- [x] https://github.com/Shopify/polaris/pull/6772
- [x] https://github.com/Shopify/polaris/pull/6786
- [x] https://github.com/Shopify/polaris/pull/6792

**What does this PR do?**

<details>
<summary>View migration</summary>

```js
import fs from 'fs';
import path from 'path';
import {fileURLToPath} from 'url';
import readmeLoader from '../.storybook/polaris-readme-loader.js';

import glob from 'glob';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const readmeFiles = glob.sync(
  path.join(__dirname, '../src/components/*/README.md'),
);

const kebabize = (str) => {
  return str
    .replace(/([a-z])([A-Z])/g, '$1-$2')
    .replace(/\s+/g, '-')
    .toLowerCase();
};

const migrateFile = async (filePath) => {
  const mdContent = fs.readFileSync(filePath, 'utf-8');
  const fileTitle = filePath.split('/').at(-2); // Button

  const fileContent = readmeLoader.call({cacheable: () => {}}, mdContent);

  fs.writeFileSync(
    path.dirname(filePath) + '/' + fileTitle + `.stories.tsx`,
    fileContent,
  );
  fs.rmSync(filePath);
};

// TODO pass the omitAppProvider and fullscreen arg from

const migrateFiles = readmeFiles.map((filePath) => migrateFile(filePath));

await Promise.all(migrateFiles);
```

</details>

1. Runs a migration of all the README.md files and converted them to `*.stories.tsx` files
1. Updates `KitchenSink` to read these files
1. Updates `main.js` to read these files
1. No regressions to a11y or visual regression test
1. Storybook works locally with no regressions
    - Removed the Playground > Playground title grouping
1. Linting errors have been resolved in `.stories.tsx` files
1. remove src/components/README.md
1. remove [build:docs step](https://github.com/Shopify/polaris/blob/8296f4304fdb72dedd17d45bc7db154bf41cc3c4/polaris-react/package.json#L52) (and from the list of steps in the build script)
1. Removes codeowners
1. Removes unused npm packages

### CODE REVIEW

**Comment by alex-page** [comment]
> Bonus work:
>
> remove src/components/README.md
> remove [build:docs step](https://github.com/Shopify/polaris/blob/8296f4304fdb72dedd17d45bc7db154bf41cc3c4/polaris-react/package.json#L52) (and from the list of steps in the build script)

File: polaris-react/tsconfig.json [hunk 1]
```diff
@@ -19,5 +19,6 @@
     "./playground",
     ".storybook/**/*.ts",
     ".storybook/**/*.tsx"
-  ]
+  ],
+  "exclude": ["./src/components/**/*.stories.tsx"]
 }
```
Comment (alex-page) [inline polaris-react/tsconfig.json:24]: This should be resolved in a follow up

File: .changeset/green-trees-beam.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
+---
+
+Bump storybook versions and clean up main.js
```
Comment (alex-page) [inline .changeset/green-trees-beam.md]:
```suggestion
'@shopify/polaris': minor
---

Remove all README.md files from components, migrate them to polaris.shopify.com and storybook examples
```

File: .changeset/green-trees-beam.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': minor
+---
+
+Remove all README.md files from components, migrate them to polaris.shopify.com and storybook examples
```
Comment (chloerice) [inline .changeset/green-trees-beam.md]:
```suggestion
Removed `README.md` files from all components and migrated their examples into Storybook files. Their documentation has moved from `polaris-react/src/components/**/README.md` to `polaris.shopify.com/content/components/*.md`. 
```

**Review by chloerice — APPROVED** [review]
Reviewed and accepted all UI Test baselines ✅  Not sure why linting is going haywire 🧐

File: polaris-react/playground/KitchenSink.tsx [hunk 4]
```diff
@@ -2,51 +2,49 @@ import React from 'react';
 
 import {RenderPerformanceProfiler} from '../.storybook/RenderPerformanceProfiler';
 
-interface ReadmeModule {
-  [key: string]: any;
+interface Stories {
+  [key: string]: React.ComponentClass;
```
Comment (BPScott) [inline polaris-react/playground/KitchenSink.tsx]: `React.ComponentType` is more correct - as that encompases the typings for function based components and class based components

```suggestion
  [key: string]: React.ComponentType;
```

File: polaris-react/scripts/readme-update-version.js [hunk 5]
```diff
@@ -9,7 +9,7 @@ const {semverRegExp} = require('../scripts/utilities');
 
 const root = resolve(__dirname, '..');
 
-const readmes = Object.freeze(['README.md', 'src/components/README.md']);
+const readmes = Object.freeze(['README.md']);
```
Comment (BPScott) [inline polaris-react/scripts/readme-update-version.js:12]: Now we're down to one item, and I can't see us adding additional files in the future, this we don't really need this to loop. Can be tidied in a follow-up.

**Review by BPScott — APPROVED** [review]
Conceptually looks great to me.

I've not reviewed every storybook story to ensure that the output remains identical, but I skimmed the first few including AppProvider which is notably edgecasey and they look good to me.

Two small non-blocking notes inline

File: polaris-react/playground/KitchenSink.tsx (hunk 4 above)
Comment (alex-page) [reply polaris-react/playground/KitchenSink.tsx]: Ahh love this. I had something there wasn't confident so I removed it. This is exactly it, cheers @BPScott .

**Pull request merged**